django/contrib/auth/forms.py

```diff
@@ -144,6 +144,23 @@ def save(self, commit=True):
         return user
 
 
+class UserCreationForm(BaseUserCreationForm):
+    """
+    Adds a method for rejecting usernames that differ only in case. Does not
+    affect existing usernames or those created via createsuperuser.
+    """
+
+    def clean_username(self):
+        username = self.cleaned_data.get("username")
+        if username and User.objects.filter(username__iexact=username).exists():
+            raise forms.ValidationError(
+                User._meta.get_field("username").error_messages["unique"],
```

Comment:
I think it would be better to declare an error message here on the form, so that when we're looking at it we don't need to go and track this down to know what it says.